cdc/sink/dmlsink/txn/mysql/mysql.go

```diff
@@ -122,29 +127,28 @@ func NewMySQLBackends(
 
 	// Inherit the default value of the prepared statement cache from the SinkURI Options
 	cachePrepStmts := cfg.CachePrepStmts
-	prepStmtCacheSize := cfg.PrepStmtCacheSize
-
-	var stmtCache *lru.Cache
 	if cachePrepStmts {
 		// query the size of the prepared statement cache on serverside
 		maxPreparedStmtCount, err := pmysql.QueryMaxPreparedStmtCount(ctx, db)
 		if err != nil {
 			return nil, err
 		}
+		if maxPreparedStmtCount == -1 {
+			// NOTE: seems TiDB doesn't follow MySQL's specification.
+			maxPreparedStmtCount = math.MaxInt
+		}
 		// if maxPreparedStmtCount == 0,
 		// it means that the prepared statement cache is disabled on serverside.
 		// if maxPreparedStmtCount/(cfg.WorkerCount+1) == 0, for each single connection,
 		// it means that the prepared statement cache is disabled on clientsize.
 		// Because each connection can not hold at lease one prepared statement.
 		if maxPreparedStmtCount == 0 || maxPreparedStmtCount/(cfg.WorkerCount+1) == 0 {
 			cachePrepStmts = false
-		} else if maxPreparedStmtCount/(cfg.WorkerCount+1) < prepStmtCacheSize {
-			// if maxPreparedStmtCount/(cfg.WorkerCount+1) < prepStmtCacheSize,
-			// it means that the prepared statement cache is too large on clientsize.
-			// adjust the size of the prepared statement cache on clientsize.
-			// to avoid error `Can't create more than max_prepared_stmt_count statements`
-			prepStmtCacheSize = maxPreparedStmtCount / (cfg.WorkerCount + 1)
 		}
+	}
+
+	var stmtCache *lru.Cache
+	if cachePrepStmts {
 		stmtCache, err = lru.NewWithEvict(prepStmtCacheSize, func(key, value interface{}) {
```

Comment:
I think we can remove the const prepStmtCacheSize, and replace it with maxPreparedStmtCount